cleanlab/datalab/internal/issue_manager/underperf_group.py

```diff
@@ -147,8 +147,7 @@ def perform_clustering(self, knn_graph: csr_matrix) -> npt.NDArray[np.int_]:
         }
         dbscan_params["metric"] = "precomputed"
         clusterer = DBSCAN(**dbscan_params)
-        clusterer.fit(knn_graph)
-        knn_graph.eliminate_zeros()  # Reset KNN Graph
+        clusterer.fit(knn_graph.copy())  # Copy to avoid modification
```

Comment:
Does `DBSCAN` modify the `knn_graph`?

Reply:
Yes. [Here](https://github.com/scikit-learn/scikit-learn/blob/2d8e03f4d5b3c466fb4542360bcef742a3a4e0a1/sklearn/cluster/_dbscan.py#L384-L389) is the code that modifies the `knn_graph`.

Reply:
Ok, let's clarify this in the comment.

```suggestion
        clusterer.fit(knn_graph.copy())  # Copy to avoid modification by DBSCAN
```

Reply:
If this is doubling the memory requirements for Datalab to run on somebody's machine, I would keep a GH Issue open to see if we can avoid this doubling. 

I wouldn't add this issue-type to the set of issues audited by default until somebody has investigated this question (can do that in a separate PR)